apis/cassandra/v1beta1/cdc_types.go

```diff
@@ -0,0 +1,50 @@
+package v1beta1
+
+// CDCConfiguration holds CDC config for the CassandraDatacenter. Note that it cannot contain arrays, channels, maps etc. because of the way the
+// reflection logic works which marshalls it into a string for the purposes of passing on the command line.
+type CDCConfiguration struct {
+	Enabled          bool    `json:"enabled"`
+	PulsarServiceUrl *string `json:"pulsarServiceUrl"`
+	// +optional
+	TopicPrefix *string `json:"topicPrefix,omitempty"`
+	// +optional
+	CDCWorkingDir *string `json:"cdcWorkingDir,omitempty"`
+	// +optional
+	CDCPollIntervalMs *int `json:"cdcPollIntervalM,omitempty"`
+	// +optional
+	ErrorCommitLogReprocessEnabled *bool `json:"errorCommitLogReprocessEnabled,omitempty"`
+	// +optional
+	CDCConcurrentProcessors *int `json:"cdcConcurrentProcessors,omitempty"`
+	// +optional
+	PulsarBatchDelayInMs *int `json:"pulsarBatchDelayInMs,omitempty"`
+	// +optional
+	PulsarKeyBasedBatcher *bool `json:"pulsarKeyBasedBatcher,omitempty"`
+	// +optional
+	PulsarMaxPendingMessages *int `json:"pulsarMaxPendingMessages,omitempty"`
+	// +optional
+	PulsarMaxPendingMessagesAcrossPartitions *int `json:"pulsarMaxPendingMessagesAcrossPartitions,omitempty"`
+	// +optional
+	PulsarAuthPluginClassName *string `json:"pulsarAuthPluginClassName,omitempty"`
+	// +optional
+	PulsarAuthParams *string `json:"pulsarAuthParams,omitempty"`
+	// +optional
+	SSLProvider *string `json:"sslProvider,omitempty"`
```

Comment:
Instead of flat structure, would it be easier for the user if there was some sort of tree?

Reply:
Can you expand on this? What struct do you envisage to encapsulate `PulsarAuthParams` and `SSLProvider`?

Reply:
For example, all SSL properties (I assume they're CDC -> target) to one struct, all Pulsar properties to one struct. If you envision that there's going to more targets than Pulsar, then next there could be "kafka" part and so on. Instead of one flat structure with hundreds of options.

Reply:
There aren't currently Kafka output capabilities in the ecosystem. I'd be open to a second nested struct for the Pulsar agent as below which would enable later additions with fewer API changes, if you wanted. 

```
type CDCConfiguration struct {
        pulsar: PulsarCDCConfiguration `...`
        ...
}

type PulsarCDCConfiguration struct {
	Enabled          bool    `json:"enabled"`
	PulsarServiceUrl *string `json:"pulsarServiceUrl"`
	// +optional
        ...
}
```

If we wanted this kind of extensibility I suspect we'd be better off with a separate CRD. However, this is planning a long way forward given there are no currently planned Kafka connectors available.

The SSL properties **are Pulsar connector properties**, so I think these would always lie in the Pulsar struct in any event.

Reply:
@burmanm - waiting on a response for this one.